src/Goteo/Repository/InvestOriginRepository.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Goteo\Repository;
+
+use Goteo\Application\Exception\ModelException;
+use Goteo\Application\Exception\ModelNotFoundException;
+use Goteo\Entity\Invest\InvestOrigin;
+use PDO;
+use PDOException;
+
+class InvestOriginRepository extends BaseRepository
+{
+    protected ?string $table = 'invest_origin';
+
+    public function getByInvestId(int $id): InvestOrigin
+    {
+        $sql = "SELECT *
+                FROM invest_origin
+                WHERE invest_origin.invest_id = ?";
+
+        $investOrigin = $this->query($sql, [$id])->fetchObject(InvestOrigin::class);
+
+        if (!$investOrigin instanceOf InvestOrigin)
+            throw new ModelNotFoundException("InvestOrigin with invest_id $id not found");
+
+        return $investOrigin;
+    }
+
+    /**
+     * @return InvestOrigin[]
+     */
+    public function getList(int $offset = 0, int $limit = 10): array
+    {
+        $sql = "SELECT *
+                FROM invest_origin
+                LIMIT $limit
+                OFFSET $offset";
+
+        return $this->query($sql)->fetchAll(PDO::FETCH_CLASS, InvestOrigin::class);
+    }
+
+    /**
+     * @return InvestOrigin[]
+     */
+    public function getListBySource(string $source): array
+    {
+        $sql = "SELECT *
+                FROM invest_origin
+                WHERE invest_origin.source = ?";
+
+        return $this->query($sql, $source)->fetchAll(PDO::FETCH_CLASS, InvestOrigin::class);
+    }
+
+    /**
+     * @return InvestOrigin[]
+     */
+    public function getListByDetail(string $detail): array
+    {
+        $sql = "SELECT *
+                FROM invest_origin
+                WHERE invest_origin.detail = ?";
+
+        return $this->query($sql, $detail)->fetchAll(PDO::FETCH_CLASS, InvestOrigin::class);
+    }
+
+    /**
+     * @return InvestOrigin[]
+     */
+    public function getListByAllocated(string $allocated): array
+    {
+        $sql = "SELECT *
+                FROM invest_origin
+                WHERE invest_origin.allocated = ?";
+
+        return $this->query($sql, $allocated)->fetchAll(PDO::FETCH_CLASS, InvestOrigin::class);
+    }
+
+    public function count(): int
+    {
+        $sql = "SELECT count(invest_origin.invest_id)
+                FROM invest_origin";
+
+        return $this->query($sql)->fetchColumn();
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getSources(): array {
+        $sql = "SELECT invest_origin.source
+                FROM invest_origin
+                GROUP BY invest_origin.source";
+
+        return $this->query($sql)->fetchAll();
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getDetails(): array {
+        $sql = "SELECT invest_origin.detail
+                FROM invest_origin
+                GROUP BY invest_origin.detail";
+
+        return $this->query($sql)->fetchAll();
+    }
```

Comment:
If detail could have repeated values, maybe it'd be interesting to have a SELECT DISTINCT(invest_origin.detail) instead.